store/subscription.go

```diff
@@ -259,15 +271,15 @@ func (s *subscription) insert(ctx context.Context, events []*model.Transfer) (er
 			}
 
 			// Add fee if it's a ETH event
-			if f, ok := fees[gethCommon.BytesToHash(e.TxHash)]; ok && bytes.Equal(e.Address, model.ETHBytes) {
+			if f, ok := fees[gethCommon.BytesToHash(e.TxHash)]; bytes.Equal(e.Address, model.ETHBytes) && ok {
 				if feeDiff[from] == nil {
 					feeDiff[from] = new(big.Int).Set(f)
 				} else {
 					feeDiff[from] = new(big.Int).Add(feeDiff[from], f)
 				}
 			}
 		}
-		if hasTo {
+		if hasTo && !bytes.Equal(e.From, RewardToMiner.Bytes()) && !bytes.Equal(e.From, RewardToUncle.Bytes()) {
```

Comment:
```
!bytes.Equal(e.From, RewardToMiner.Bytes()) && !bytes.Equal(e.From, RewardToUncle.Bytes())
```
This case should never happen because your `RewardToMiner` and `RewardToUncle` is not a standard address, so `hasTo` is always false.

Reply:
這邏輯我想確保只有轉入的event會被處理。
如果是獎勵，那就不做（留給 [這裡](https://github.com/getamis/eth-indexer/pull/28/files/4c9e73f2c8f593e79a852b73b6b9f60cceacb09e#diff-512c955ca30400625a29ab1219dc27f5R293)處理），好處是最後可以知道miner/uncle reward。

另一個原因是用來阻擋重複紀錄兩次balance
如果一個block coinbase剛好是訂閱帳號之一，則balance被重複計算(1是[這裡](https://github.com/getamis/eth-indexer/pull/28/files/4c9e73f2c8f593e79a852b73b6b9f60cceacb09e#diff-512c955ca30400625a29ab1219dc27f5R285), 2是 [這裡](https://github.com/getamis/eth-indexer/pull/28/files/4c9e73f2c8f593e79a852b73b6b9f60cceacb09e#diff-512c955ca30400625a29ab1219dc27f5R295))

當然我也可以讓reward變成是event之一(也就是 [移除minerRewardDiff](https://github.com/getamis/eth-indexer/pull/28/files/4c9e73f2c8f593e79a852b73b6b9f60cceacb09e#diff-512c955ca30400625a29ab1219dc27f5R243))，但最後就會不知道有多少$$來自miner/uncles


判斷miner/uncle reward的[邏輯](https://github.com/getamis/eth-indexer/pull/28/files/4c9e73f2c8f593e79a852b73b6b9f60cceacb09e#diff-512c955ca30400625a29ab1219dc27f5R293)我可以往hasTo 裡面移，但就是會變很多if在裡面 😆 

如果有更好的寫法也請提供 xd